ui/pages/confirmations/confirmation/confirmation.js

```diff
@@ -497,20 +509,31 @@ export default function ConfirmationPage({
           >
             <Icon name={IconName.ArrowRight} />
           </button>
-        </div>
+        </Box>
+      )}
+      {isSnapCustomUIDialog && (
```

Comment:
Not introduced by this PR, but I'm not a fan of coupling to specific approval types. The nature of this page is to be a generic template for confirmations, meaning if we want to generalise something, we should do it based off new properties in each approval template such as `showSnapHeader: true` for example.

Reply:
I agree up to some point. It would be even better maybe if we have some kind of Snap component that we just plug in and display conditionally (one that would render and handle all Snap UI required). That way this logic would not be required anymore and I would maintain only Snap component instead of this file. The major obstacles to that are headers/footers I think from UI perspective 🤔 

By making designs consistent and sticky header in this PR, we're one step closer to having something like that.
We would need complete refactoring of this file for sure.

This PR unblocks some stuff for us, I would like to go with it for now.